src/app/dashboard-widgets/create-work-item-widget/create-work-item-widget.component.ts

```diff
@@ -1,54 +1,77 @@
+  WorkItemsData
+} from '../../shared/workitem-utils';
 
 @Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
   selector: 'fabric8-create-work-item-widget',
   templateUrl: './create-work-item-widget.component.html'
 })
 export class CreateWorkItemWidgetComponent implements OnInit {
 
   @Input() userOwnsSpace: boolean;
+  @Input() loggedInUser: User;
+  @Input() currentSpace: Space;
 
-  private _myWorkItems: ConnectableObservable<WorkItem[]>;
+  private _myWorkItems: Observable<WorkItem[]>;
   myWorkItemsCount: Observable<number>;
   contextPath: Observable<string>;
 
   constructor(
     private filterService: FilterService,
     private workItemService: WorkItemService,
-    private userService: UserService,
     private contexts: Contexts
   ) {}
 
   ngOnInit() {
     this.contextPath = this.contexts.current.pipe(map(context => context.path));
-    this._myWorkItems = this.userService
-      .getUser().pipe(
-      tap(() => this.workItemService.buildUserIdMap()),
-      switchMap(() => this.userService.loggedInUser),
-      map(user => this.filterService.queryBuilder('assignee', this.filterService.equal_notation, user.id)),
-      switchMap(filters => this.workItemService
-        .getWorkItems(100000, {expression: filters})),
-      map((val: WorkItemsData) => val.workItems),
-      map(workItems => filterOutClosedItems(workItems)),
-      // Resolve the work item type
-      tap(workItems => workItems.forEach(workItem => this.workItemService.resolveType(workItem))),
-      tap(workItems => workItems.forEach(workItem => this.workItemService.resolveAreaForWorkItem(workItem))),
-      // MUST DO creator after area due to bug in planner
-      tap(workItems => workItems.forEach(workItem => this.workItemService.resolveCreator(workItem))),
-      publishReplay(1)) as ConnectableObservable<WorkItem[]>;
-    this._myWorkItems.connect();
-    this.myWorkItemsCount = this._myWorkItems.pipe(map(workItems => workItems.length));
+    this.getCurrentSpaceWorkItems();
+  }
+
+  ngOnChanges() {
+    this.getCurrentSpaceWorkItems();
   }
 
   get myWorkItems(): Observable<WorkItem[]> {
     return this._myWorkItems;
   }
 
+  getCurrentSpaceWorkItems() {
+    let filters = {};
+    [
+      buildAssigneeQuery(this.filterService, this.loggedInUser.id),
+      buildSpaceQuery(this.filterService, this.currentSpace.id),
+      buildClosedWorkItemQuery(this.filterService)
+    ].forEach(query => {
+      filters = this.filterService.queryJoiner(
+        filters, this.filterService.and_notation, query
+      );
+    });
+
+    this._myWorkItems = this.workItemService
+      .getWorkItems(100000, {expression: filters}).pipe(
+        map((val: WorkItemsData) => val.workItems),
+        tap(workitems => console.log(workitems)),
+        // Resolve the work item type
+        tap((workItems: WorkItem[]) => workItems.forEach((workItem: WorkItem) => this.workItemService.resolveType(workItem))),
```

Comment:
should have a `: void` for the inner fn of these `tap`s similar to the one above for `map`